### ISSUE TITLE
Add tests for precedence and associativity?

### ISSUE THREAD

**Issue by eed3si9n** [eclipse-biscuit/tree-sitter-biscuit#2]
https://doc.biscuitsec.org/reference/specifications

Given that binary_op_expression is the central point of complexity, it might be good to add tests for corner cases to test operator precedence so if we want, we can start refactoring rules around it?

**Comment by divarvel** [comment]
yes, there is a shocking lack of tests for now (operator precedence / assoc is tested a bit here: https://github.com/biscuit-auth/tree-sitter-biscuit/blob/main/test/corpus/rules-expr.txt#L5 )

I have added some in https://github.com/biscuit-auth/tree-sitter-biscuit/pull/1 but more would be welcome.

Before refactoring things, the first objective would be proper parsing of method calls, which are conflicting with predicates (it’s not a conflict detected by tree-sitter ~~, but method calls are identified as predicates during parsing~~)

**Comment by eed3si9n** [comment]
> Before refactoring things...

Could you create an issue for that plz? A repro example would be helpful.

**Comment by divarvel** [comment]
https://github.com/biscuit-auth/tree-sitter-biscuit/issues/3 showcases the issue in the v5 branch

**Comment by divarvel** [comment]
i have extracted tests from the sample suite and added them in #6
